buildSrc/src/main/kotlin/ai/koog/gradle/plugins/CheckSplitPackagesPlugin.kt

```diff
@@ -0,0 +1,220 @@
+                                .toSet() // de-dup per JAR
+                                .forEach { pkg ->
+                                    packagesToJars.getOrPut(pkg) { linkedSetOf() }.add(jarFile)
+                                }
+                        }
+                    }
+
+                    // Build a quick index of jar -> owning subproject path if jar lies under a buildDir
+                    val allProjects = project.rootProject.allprojects
+                    fun ownerOf(jar: java.io.File): String? {
+                        val abs = jar.absolutePath
+                        return allProjects.firstOrNull { p -> abs.startsWith(p.layout.buildDirectory.asFile.get().absolutePath + java.io.File.separator) }?.path
+                    }
+
+                    val split = packagesToJars
+                        .mapValues { (_, jars) -> jars.map { it to (ownerOf(it)) } }
+                        .filterValues { pairs ->
+                            pairs.map { it.first.absolutePath }.toSet().size > 1
+                        } // distinct jar files > 1
+                        .toSortedMap()
+
+                    if (split.isNotEmpty()) {
+                        val affected = split.keys.joinToString()
+                        if (failOnError) {
+                            logger.error("Detected split packages in ${project.path} (${split.size} packages): $affected")
+                            printSplitDetails(logger, split, warnMode = false)
+                            throw GradleException("️😱 Split packages detected in ${project.path}.")
+                        } else {
+                            logger.warn("Detected split packages in ${project.path} (${split.size} packages): $affected")
+                            printSplitDetails(logger, split, warnMode = true)
+                            logger.warn("⚠️ Split packages detected in ${project.path}, but failOnError=false. See details above.")
+                        }
+                    } else {
+                        logger.info("✅ No split packages detected in ${project.path}.")
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun printSplitDetails(
+    logger: org.gradle.api.logging.Logger,
+    split: Map<String, List<Pair<java.io.File, String?>>>,
+    warnMode: Boolean
+) {
+    split.forEach { (pkg, jarPairs) ->
+        val logLevel = if (warnMode) LogLevel.WARN else LogLevel.ERROR
+        logger.log(logLevel, "- $pkg")
+        jarPairs.map { it.first }.sortedBy { it.name }.forEach { jf ->
+            val owner = jarPairs.firstOrNull { it.first == jf }?.second
+            if (owner != null) {
+                logger.log(logLevel, "    in: ${jf.name}  (project: $owner)")
+            } else {
+                logger.log(logLevel, "    in: ${jf.name}")
+            }
+        }
+    }
+}
+
+/**
+ * Kotlin DSL convenience to both apply the plugin and configure its extension in one call.
```

Comment:
I would generally prefer not to do that. It's better apply plugins more, it would make it easier to understand at a glance which plugins are currently applied. And it's only one line of code anyways, not that much of a saving here.